### PR TITLE
Set power equation for reference frame water

### PR DESCRIPTION
**Pull request by willirath**

### CODE REVIEW

File: src/rasmus_fuel/realistic.py [hunk 1]
```diff
@@ -126,11 +126,25 @@ def power_maintain_sog(
...
+    resistance_x = wave_resistance_x + water_resistance_x + wind_resistance_x
+    resistance_y = wave_resistance_y + water_resistance_y + wind_resistance_y
+
+    power_needed = resistance_x * u_ship_tw + resistance_y * v_ship_tw
+    
+    # Reference frame Earth's observer
+    #power_needed = (
+    #    coeff_water_drag * (speed_tw ** 2) * speed_og
+    #    + coeff_wind_drag * speed_rel_to_wind ** 2 * speed_og
+    #    + coeff_wave_drag * speed_og ** 2
```
Comment (willirath) [inline src/rasmus_fuel/realistic.py:147]:
```suggestion
```

File: src/rasmus_fuel/realistic.py [hunk 2]
```diff
@@ -98,6 +98,19 @@ def power_maintain_sog(
...
+    speed_wind = (u_wind ** 2 + v_wind ** 2) ** 0.5
+
+    # calc angle between wind and speed through water 
+    angle_ship_wind = np.arccos(
+        (u_wind * u_ship_tw  + v_wind * v_ship_tw) / speed_tw / wind_speed
+    )
+
+    # calc projection of ship velocity through water on wind components
+    u_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.cos(angle_wind)
+    v_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.sin(angle_wind)
```
Comment (willirath) [inline src/rasmus_fuel/realistic.py:113]: Not sure about this. Shouldn't this just be the eastward and northward components of inverse of the the apparent wind? So then, we'd have: 
```python
u_speed_rel_to_wind = u_ship_og - u_wind
v_speed_rel_to_wind = v_ship_og - v_wind
speed_rel_to_wind = (u_speed_rel_to_wind ** 2 + u_speed_rel_to_wind ** 2) ** 0.5
```

**Comment by willirath** [comment]
@eshchekinova 

- tests are all failing. Can you check these?
- linter complains. Please run `pre-commit`
- See https://github.com/willirath/rasmus_fuel/pull/13#discussion_r773979550

**Comment by eshchekinova** [comment]
The errors were originated in the realistic.py, they are corrected, now the pytest is running successfully.

File: src/rasmus_fuel/realistic.py [hunk 3]
```diff
@@ -95,10 +95,27 @@ def power_maintain_sog(
...
+        np.arccos(
+            (u_wind * u_ship_tw + v_wind * v_ship_tw) / (speed_tw + 10e-6) / (speed_wind + 10e-6)
+        ),
+        0,
+    )
+
+    # calc projection of ship velocity through water on wind components
+    u_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.cos(angle_wind)
+    v_speed_rel_to_wind = speed_tw * np.cos(angle_ship_wind) * np.sin(angle_wind)
+
```
Comment (willirath) [inline src/rasmus_fuel/realistic.py:118]: I think this should just be:
```suggestion
    u_speed_rel_to_wind = u_ship_og - u_wind
    v_speed_rel_to_wind = v_ship_og - v_wind
```

The strength and direction of the drag force is evaluated from the relative movement of the air across the vessel. So we just take the difference between the ship velocity vector and the wind velocity vector in any reference system. As the wind is given over ground, we also use ship velocity over ground.

Note that for the conversion to power, we still need to multiply with speed through water as this is the reference system relevant for the propulsion setup.

**Comment by willirath** [comment]
@eshchekinova There's only one thing left to do here: See this comment https://github.com/willirath/rasmus_fuel/pull/13/files#r774563700 for simplification of the x and y component of apparent wind / ship movement against the wind.

File: src/rasmus_fuel/realistic.py (hunk 3 above)
Comment (eshchekinova) [reply src/rasmus_fuel/realistic.py:118]: Yes, clear. For the wind-ship we do use reference with respect to Earth observer. I was confused.

**Comment by willirath** [comment]
Looks good. I'm merging.

**Pull request merged**